classes/ramen_shop/RamenShopManager.php

```diff
@@ -0,0 +1,441 @@
+    const BASIC_RAMEN_LARGE = "SHIO_L";
+    const SPECIAL_RAMEN_SHOYU = "SHOYU";
+    const SPECIAL_RAMEN_KING = "KING";
+    const SPECIAL_RAMEN_SPICY_MISO = "SPICY_MISO";
+    const SPECIAL_RAMEN_WARRIOR = "WARRIOR";
+
+    /* base ramen costs indexed by ramen key */
+    const BASE_RAMEN_COSTS = [
+        self::BASIC_RAMEN_SMALL => 5,
+        self::BASIC_RAMEN_MEDIUM => 25,
+        self::BASIC_RAMEN_LARGE => 50,
+        self::SPECIAL_RAMEN_SHOYU => 1250,
+        self::SPECIAL_RAMEN_KING => 2500,
+        self::SPECIAL_RAMEN_SPICY_MISO => 5000,
+        self::SPECIAL_RAMEN_WARRIOR => 10000,
+    ];
+
+    /* ramen effect descriptions indexed by ramen key */
+    const RAMEN_EFFECT_DESCRIPTIONS = [
+        self::SPECIAL_RAMEN_SHOYU => "+2 Stealth.",
+        self::SPECIAL_RAMEN_KING => "+1 Reputation gain (PvP excluded).",
+        self::SPECIAL_RAMEN_SPICY_MISO => "+25% base regen.",
+        self::SPECIAL_RAMEN_WARRIOR => "Heal to full after winning a battle",
+    ];
+
+    const MYSTERY_RAMEN_COST_MULTIPLIER = 1;
+    const MYSTERY_RAMEN_DURATION_MULTIPLIER = 5;
+    const BASE_RAMEN_DURATION_MINUTES = 10;
+    const BASE_MYSTERY_RAMEN_CHANCE = 5;
+    const MYSTERY_RAMEN_CHANCE_PER_PURCHASE = 5;
+
+    /**
+     * Load the ramen shop owner details for the given village
+     * @param Village $village
+     * @return RamenOwnerDto
+     */
+    public static function loadRamenOwnerDetails(Village $village): RamenOwnerDto {
+        return new RamenOwnerDto(
+            name: self::RAMEN_SHOP_OWNER_NAMES_BY_VILLAGE[$village->village_id],
+            image: self::RAMEN_SHOP_OWNER_IMAGES_BY_VILLAGE[$village->village_id],
+            background: self::RAMEN_SHOP_OWNER_BACKGROUNDS_BY_VILLAGE[$village->village_id],
+            shop_description: self::RAMEN_SHOP_DESCRIPTIONS_BY_VILLAGE[$village->village_id],
+            dialogue: self::RAMEN_SHOP_DIALOGUE_OPTIONS_BY_VILLAGE[$village->village_id][array_rand(self::RAMEN_SHOP_DIALOGUE_OPTIONS_BY_VILLAGE[$village->village_id])],
+            shop_name: self::RAMEN_SHOP_NAMES_BY_VILLAGE[$village->village_id],
+        );
+    }
+
+    /**
+     * Load the basic ramen details for the given player
+     * @param System $system
+     * @param User $player
+     * @return BasicRamenDto[]
+     */
+    public static function loadBasicRamen(System $system, User $player): array {
+        $ramen_cost_multiplier = 1 - ($player->village->active_upgrade_effects[VillageUpgradeConfig::UPGRADE_EFFECT_RAMEN_COST] / 100);
+        $rankManager = new RankManager($system);
+        $rankManager->loadRanks();
+
+        $health[1] = $rankManager->healthForRankAndLevel(1, $rankManager->ranks[1]->max_level);
+        $health[2] = $rankManager->healthForRankAndLevel(2, $rankManager->ranks[2]->max_level);
+        $health[3] = $rankManager->healthForRankAndLevel(3, $rankManager->ranks[3]->max_level);
+        $health[4] = $rankManager->healthForRankAndLevel(4, $rankManager->ranks[4]->max_level);
```

Comment:
Hmm, I think this should work - Doesn't look like anywhere we're using the non-player-rank values of `$health`
```suggestion
       $player_rank_max_health = $rankManager->healthForRankAndLevel(
         $player->rank_num, 
         $rankManager->ranks[$player->rank_num]->max_level
       );
```